### PR TITLE
fix(element-template): unlink template on replace

### PR DESCRIPTION
**Pull request by azeghers**
__Which issue does this PR address?__

Related to https://github.com/camunda/camunda-modeler/issues/1961

__Acceptance Criteria__

* [ ] Corresponds to the concept 
* [ ] Corresponds to the design 

__Definition of Done__

* [ ] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [ ] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [ ] passes Continuous Integration checks
* [ ] available as feature branch on GitHub
  * [ ] contains the cleaned up commit history
  * [ ] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [ ] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    // when
+    var newElement = bpmnReplace.replaceElement(task, newElementData);
+
+    // then
+    var businessObject = newElement.businessObject;
+    expect(businessObject.modelerTemplate).to.exist;
+  }));
+
+  it('should unlink the template', inject(function(elementRegistry, bpmnReplace) {
```
Comment (pinussilvestrus) [inline test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:44]: Always make sure we have two new lines between each test

```js
it('should keep the template linked');


it('should unlink the template');
```

Comment (azeghers) [reply test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:45]: Thanks, fixed!

**Review by pinussilvestrus — APPROVED** [review]
Changes look good to me. 

One minor thing: the linting currently does not consider missing new lines before comments (cf. see [this line of code in this PR](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/392/files#diff-72166361aad6502802da1803a7dd91291f43bcf6d1119d370193ec2d12ac53c2R46)). That's a general thing inside this project, I just opened a new issue for this: https://github.com/bpmn-io/bpmn-js-properties-panel/issues/393

File: test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+var bootstrapModeler = require('../../../../helper').bootstrapModeler,
+    camundaModdlePackage = require('camunda-bpmn-moddle/resources/camunda'),
+    coreModule = require('bpmn-js/lib/core').default,
+    inject = require('../../../../helper').inject,
+    modelingModule = require('bpmn-js/lib/features/modeling').default,
+    propertiesPanelModule = require('lib'),
+    propertiesProviderModule = require('lib/provider/camunda'),
+    replaceModule = require('bpmn-js/lib/features/replace').default;
+
+describe('element-templates - replace behavior', function() {
```
Comment (pinussilvestrus) [inline test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:12]: One thing we could consider before merging it: 

adding a test case that verifies it keeps also applied for a sub-type. E.g. when changing from `bpmn:Task` to `bpmn:ServiceTask`.

Comment (azeghers) [reply test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:12]: Good point, I'll take care of it 👍

Comment (azeghers) [reply test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:12]: Do you have better test names than those?
```
  element-templates - replace behavior
    ✓ should link to an element of a type the template applies to
    ✓ should link to a sub-element of a type the template applies to
    ✓ should not link to an element of a type the template doesn't apply to
```

Comment (pinussilvestrus) [reply test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:12]: What about

```
* should keep the template applied
* should keep the template applied - parent type
* should unlink the template if not applicable
```

Comment (philippfromme) [reply test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:12]: * should not unlink if template can be applied to type
* should not unlink if template can be applied to parent type
* should unlink if template cannot be applied to type

File: test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  ];
+
+  var diagramXML = require('./fixtures/replace-behavior.bpmn');
+
+  beforeEach(bootstrapModeler(diagramXML, {
+    elementTemplates: require('./fixtures/replace-behavior'),
+    modules : testModules,
+    moddleExtensions: { camunda: camundaModdlePackage }
+  }));
+
```
Comment (philippfromme) [inline test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:28]: Add another new line here.

File: test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      type: 'bpmn:SubProcess'
+    };
+
+    // when
+    var newElement = bpmnReplace.replaceElement(task, newElementData);
+
+    // then
+    var businessObject = newElement.businessObject;
+    expect(businessObject.modelerTemplate).not.to.exist;
+  }));
```
Comment (philippfromme) [inline test/spec/provider/camunda/element-templates/ReplaceBehaviorSpec.js:77]: Add new line here. As a general rule, we add two new lines between `it` blocks and between setup blocks and `it` blocks.

```javascript
setup();

moreSetup();


describe('...', function() {

  setup();

  moreSetup();


  it('should ...');


  it('should ...');


  describe('...', function() {

    it('should ...');

  });

});
```

File: test/spec/provider/camunda/element-templates/fixtures/replace-behavior.bpmn [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:camunda="http://camunda.org/schema/1.0/bpmn" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:camunda="http://camunda.org/schema/1.0/bpmn" id="Definitions_1" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="1.13.0-dev">
+  <bpmn:process id="Process_1" isExecutable="false">
+    <bpmn:task id="task_1" name="task_1" camunda:modelerTemplate="replace_1">
```
Comment (philippfromme) [inline test/spec/provider/camunda/element-templates/fixtures/replace-behavior.bpmn:4]: Change to `Task_1`.

**Review by philippfromme — CHANGES_REQUESTED** [review]
Almost there.

**Review by philippfromme — APPROVED** [review]
👏

**Comment by pinussilvestrus** [comment]
Good stuff 👍

**Pull request merged**